### ISSUE TITLE
train_from_iterator does not work when total length of iterable > ~4500

### ISSUE THREAD

**Issue by dsaks999** [huggingface/tokenizers#906]
The following code, where the total length of the array (itr) being passed to train_from_iterator is 3880536, fails when I either pass the array straight to train_from_iterator or define a generator (as shown below). The error states that None type is not iterable.

`bert_tokenizer = Tokenizer(WordPiece(unk_token="[UNK]"))

bert_tokenizer.normalizer = normalizers.Sequence([NFD(), Lowercase(), StripAccents()])

# bert_tokenizer.pre_tokenizer = BertPreTokenizer()
bert_tokenizer.pre_tokenizer = Whitespace()

bert_tokenizer.post_processor = TemplateProcessing(
    single="[CLS] $A [SEP]",
    pair="[CLS] $A [SEP] $B:1 [SEP]:1",
    special_tokens=[
        ("[CLS]", 1),
        ("[SEP]", 2),
    ],
)

trainer = WordPieceTrainer(
  special_tokens=["[UNK]", "[CLS]", "[SEP]", "[PAD]", "[MASK]"]
)

itr = df_activity_memo_npt_activity_global.Activity_Memo.values

def batch_iterator(dataset=itr, batch_size=100):
  for i in range(0, len(dataset), batch_size):
    yield dataset[i: i + batch_size]
  
bert_tokenizer.train_from_iterator(batch_iterator(), trainer=trainer, length=len(itr))`

When I pass the array with a length of 4500, as shown below, the code runs without error, both with and without a generator.

`bert_tokenizer = Tokenizer(WordPiece(unk_token="[UNK]"))

bert_tokenizer.normalizer = normalizers.Sequence([NFD(), Lowercase(), StripAccents()])

# bert_tokenizer.pre_tokenizer = BertPreTokenizer()
bert_tokenizer.pre_tokenizer = Whitespace()

bert_tokenizer.post_processor = TemplateProcessing(
    single="[CLS] $A [SEP]",
    pair="[CLS] $A [SEP] $B:1 [SEP]:1",
    special_tokens=[
        ("[CLS]", 1),
        ("[SEP]", 2),
    ],
)

trainer = WordPieceTrainer(
  special_tokens=["[UNK]", "[CLS]", "[SEP]", "[PAD]", "[MASK]"]
)

itr = df_activity_memo_npt_activity_global.Activity_Memo.values[0:4500]
print(len(itr))

def batch_iterator(dataset=itr, batch_size=100):
  for i in range(0, len(dataset), batch_size):
    yield dataset[i: i + batch_size]
  
bert_tokenizer.train_from_iterator(batch_iterator(), trainer=trainer, length=len(itr))`

Is there a particular limit with train_from_iterator?

**Comment by dsaks999** [comment]
It looks as though I actually had some NA values in that array `df_activity_memo_npt_activity_global.Activity_Memo.values[0:4500]` which were probably causing the error. Thanks.
